File: back/src/main/java/com/back/domain/book/dto/BookWithTagDto.java
package com.back.domain.book.dto;

import com.back.domain.book.entity.Book;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public record BookWithTagDto(
        @NotNull
        Long id,
        @NotNull
        String title,
        @NotNull
        String imgUrl,
        @NotNull
        Double averageRating,
        @NotNull
        List<String> tags
) {
    public BookWithTagDto(Book book, List<String> tags) {
        this(
                book.getId(),
                book.getTitle(),
                book.getImgUrl(),
                book.getAverageRating(),
                tags
        );
    }
}

File: back/src/main/java/com/back/domain/book/dto/BookWithTagDto.kt
package com.back.domain.book.dto

import com.back.domain.book.entity.Book
import jakarta.validation.constraints.NotNull

class BookWithTagDto(
    val id: Long,
    val title: String,
    val imgUrl: String?,
    val averageRating: Double,
    val tags: List<String>
) {
    constructor(book: Book, tags: List<String>) : this(
        book.id,
        book.title,
        book.imgUrl,
        book.averageRating,
        tags
    )
}
